### DISCUSSION TITLE
Question: would there be an interest in incorporating my termios bindings into the standard library?

### DISCUSSION THREAD

**Discussion by strawedberries** [c3lang/c3c#1499] [General]
Hello dear C3 community!

I noticed that the termios bindings were rather lacking, so I spent some time writing my own bindings for FreeBSD's "termios.h" and "_termios.h" header files, into a single "termios.c3" file. 

Once I've cleaned up the code, would there be an interest in reviewing and potentially incorporating those bindings into the standard library so others can benefit from it as well?

Some things that I should probably mention, as it is important to know:
- I have never worked with GitHub before, and rarely worked with `git`, so I am not sure how I would go about contributing and will likely make mistakes along the way.
- I have only started programming in C3 yesterday, so I am mostly unfamiliar with the language.
- I am not sure if bindings are accepted as it is not "pure C3", but I do definitely see bindings in the standard library.
- The bindings as of right now are very crudely done and as such lack including, but not limited to:
    - Proper checks for the environment, amongst other checks.
    - Usage of proper types (CInt, etc).
    - Documentation.

**Comment by Caleb-o** [comment]
o7

Not sure about standard library, but it could definitely be possible for the vendor I would think. Unless Lerno would think it would be better in the standard?

**Reply by strawedberries** [reply]
> Not sure about standard library

Ah, my bad. I didn't necessarily mean the standard library but moreso adding it to the project's available modules in general, wherever that would be most appropriate. I couldn't really find the proper words to express that.

**Reply by Caleb-o** [reply]
That's all good! Have a look at [vendor](https://github.com/c3lang/vendor). There's some rules to make sure the library fits naming conventions etc, but I believe this should be a good place to PR/chat about getting termios in. If it gets in, you can then use the `c3c vendor-fetch` command to pull the package. If you can get it following the guidelines, then it shouldn't be a problem I don't think.

**Reply by strawedberries** [reply]
I'll have a look at it right away! Do expect this to take a little while though due to the before-mentioned complications like lack of experience with the language.

**Reply by Caleb-o** [reply]
All is well. If you have any questions, feel free to join the [Discord](https://discord.gg/qN76R87) and we'll do our best to help!
